rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/StateDedupedJobScheduler.java

```diff
@@ -61,20 +63,29 @@ public boolean schedule(final String tube, final String jobStr, final int runInM
     public boolean deleteJob(final String tube, String jobStr) {
         try (Handle handle = rdbi.open()) {
             return 1 == handle.attach(StateDedupedJobSchedulerDAO.class)
-                              .deleteJob(getReadyQueue(tube), getRunningQueue(tube), jobStr);
+                              .deleteJob(getReadyQueue(tube),
+                                         getRunningQueue(tube),
+                                         getReadyAndRunningQueue(tube),
+                                         jobStr);
         }
     }
 
     public boolean deleteJobFromReady(final String tube, String jobStr) {
         try (Handle handle = rdbi.open()) {
-            return 1 == handle.jedis().zrem(getReadyQueue(tube), jobStr);
+            return 1 == handle.attach(StateDedupedJobSchedulerDAO.class)
+                              .deleteJobFromReady(getReadyQueue(tube),
+                                                  getReadyAndRunningQueue(tube),
+                                                  jobStr);
         }
     }
 
     public boolean ackJob(final String tube, String jobStr) {
         try (Handle handle = rdbi.open()) {
             return 1 == handle.attach(StateDedupedJobSchedulerDAO.class)
-                              .ackJob(getRunningQueue(tube), jobStr);
+                              .ackJob(getReadyQueue(tube),
```

Comment:
shouldn't do it from ready queue if we have a second job with the same id waiting to be run. this could happen for cases like multiple fb edits of same post.